xbmc/storage/windows/Win32StorageProvider.cpp

```diff
@@ -96,6 +98,7 @@ CDetectDisc::CDetectDisc(const std::string &strPath, const bool bautorun)
 bool CDetectDisc::DoWork()
 {
   CLog::Log(LOGDEBUG, "%s: Optical media found in drive %s", __FUNCTION__, m_strPath.c_str());
+#ifdef HAS_DVD_DRIVE
```

Comment:
I'll update to ifdef away the complete `CDetectDisc` class.

`g_mediaManager.GetDiskLabel` and `g_mediaManager.GetDiskUniqueId` are the methods missing when `HAS_DVD_DRIVE` is undefined.